framework/src/base/MooseApp.C

```diff
@@ -1160,7 +1166,8 @@ MooseApp::registerRestartableData(const std::string & name,
                                   bool read_only)
 {
   // Select the data store for saving this piece of restartable data (mesh or everything else)
-  auto & data_ref = mesh_meta_data ? _mesh_meta_data_map : _restartable_data[tid];
+  auto & data_ref =
+      mesh_meta_data ? _restartable_meta_data.at("MeshMetaData") : _restartable_data[tid];
```

Comment:
I'm not a huge fan of using at(). Note that you get a really nasty message when running in MPI since this API won't call MPI_Abort().

Reply:
What is the alternative to avoid accidental insertion, because calling _restartable_meta_data["foo"] can create an empty RestartableDataMap. Do I used find and error if it comes back null?

Reply:
use .find()

Reply:
I used find.